metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
The reason I had previously suggested polymorphism is so that the children of BasicParser and the children of GrokParser would not have apparently different interfaces.  Granted making both implement both, when each only wants one, is less than ideal.  But I think the suggestion I just sent is best: no need to change the interface.